### DISCUSSION TITLE
Failed to publish boardman.ipa to App Store Connect.

### DISCUSSION THREAD

**Discussion by inioluwadavid** [codemagic-ci-cd/codemagic-docs#2785] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I. The code built successfully but failed to publish to appstore 
<img width="1337" alt="Screenshot 2024-10-14 at 7 34 42 AM" src="https://github.com/user-attachments/assets/bb7754ed-907b-4406-921a-a3403698c99b">

### Expected results

Build published sucessfully to appstore

### Actual results

<img width="1337" alt="Screenshot 2024-10-14 at 7 34 42 AM" src="https://github.com/user-attachments/assets/92dbd9e8-ca75-458a-bfc6-a29c8bd14f9c">

### Build id (optional)

235

**Comment by jimsakiko** [comment]
Hello @inioluwadavid , 

I don't have the fix for you, but I noticed that your log shows Bitcode usage, which is no longer acceptable for iOS after Xcode 16.  See here:
https://medium.com/@abdulahad2024/fixing-bitcode-issues-in-xcode-16-how-to-resolve-invalid-executable-errors-when-uploading-ios-da07a5a39c7c

**Comment by dam-ease** [comment, marked as answer]
Hello @inioluwadavid,
This issue arises from 3rd party dependencies on iOS that still rely on bitcode. One option is to remain on Xcode 15.4, while another solution involves stripping bitcode from the 3rd party frameworks. You can refer to [this article](https://medium.com/@abdulahad2024/fixing-bitcode-issues-in-xcode-16-how-to-resolve-invalid-executable-errors-when-uploading-ios-da07a5a39c7c) for an example.
